index.bs

```diff
@@ -5952,6 +5962,142 @@ However, [=authenticators=] that do not utilize [[!FIDO-CTAP]] do not necessaril
+        ; to be the same between an app and a browser on the same device, while
+        ; an "app"-scoped key would probably not be.
+        ;
+        ; Whatever the scope, a device key is still specific to a given credential
+        ; and does not provide any ability to link credentials.
+        ;
+        ; Whether device-scoped or not, keys are still device-bound. I.e. an
+        ; app-scoped key does not enjoy lesser protection from extraction.
+
+        scope: uint .size 1, ; a value of 0x00 means "entire device" ("all apps") scope.
+                             ; 0x01 means "per-app" scope.
+                             ; Values other than 0x00 or 0x01 are reserved for future use.
+
+        ; See https://www.w3.org/TR/webauthn/#sctn-generating-an-attestation-object
+        ;
+        ; Attestation statement formats define the \`fmt\` and \`attStmt\` members of
+        ; $$attStmtType.
+        ;
+        ; In summary, the \`attStmt\` will (typically) contain:
+        ;   (1) a SIGNATURE value calculated (using the attestation private key)
+        ;       over (aaguid || dpk).
+        ;   (2) the attestation certificate or public key, and supporting certificates,
+        ;       if any.
+        ;
+        ; Note that there are details dependent upon the particular attestation
+        ; statement format.
+        ; See https://www.w3.org/TR/webauthn/#sctn-defined-attestation-formats.
+
+        $$attStmtType,
+    }
+
+    ```
+
+: Authenticator extension processing
+:: For both [=authenticatorMakeCredential=] and [=authenticatorGetAssertion=] operations:
+    1. Create or select the [=user credential=] as usual, and let `userCredentialId` be its <code>[=credentialId=]</code>.
+
+    1. If a [=hardware-bound device key pair=] does not already exist for this [=user credential=] on the [=authenticator=], create it using the same public key algorithm as that used by the [=user credential=]'s [=credential key pair=], otherwise locate the existing [=device-bound key=].
+
+    1. Let `dpk` be the newly created or existing [=device public key=], in COSE_Key format [=credentialPublicKey|in the same fashion as for the user credential's credentialPublicKey=] when the latter is conveyed in [=attested credential data=].
+
+    1. Let `devicePrivateKey` be the newly created or existing [=device private key=].
+
+    1. Let `clientDataHash` be the [=hash of the serialized client data=].
+
+    1. Let `sig` be the result of signing the concatenation of `clientDataHash` and `userCredentialId` using the `devicePrivateKey` and the signature algorithm appropriate for the `devicePrivateKey`'s public key algorithm: `sign((clientDataHash || userCredentialId), devicePrivateKey)`.
+
+    1. Let `scope` have the value zero (0x00) if this is an "entrie-device" [=device public key=]. Otherwise, let `scope` have the value one (0x01), indicating a more narrow per-app scope.
```

Comment:
```suggestion
    1. Let `scope` have the value zero (0x00) if this is an "entire-device" [=device public key=]. Otherwise, let `scope` have the value one (0x01), indicating a more narrow per-app scope.
```
```suggestion
    1. Let `scope` have the value zero (0x00) if this is an "entrie-device" [=device public key=]. Otherwise, let `scope` have the value one (0x01), indicating a more narrow per-app scope.
```
```suggestion
    1. Let `scope` have the value zero (0x00) if this is an "entrie-device" [=device public key=]. Otherwise, let `scope` have the value one (0x01), indicating a more narrow per-app scope.
```